network/fabrics_test.go

```diff
@@ -0,0 +1,875 @@
+package network_test
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"strings"
+	"testing"
+
+	"strconv"
+
+	"io/ioutil"
+
+	"github.com/joyent/triton-go/network"
+	"github.com/joyent/triton-go/testutils"
+	"github.com/pkg/errors"
+)
+
+var (
+	fakeFabricVlanId      = 2
+	deleteFabricErrorType = errors.New("unable to delete fabric")
+	deleteVLanErrorType   = errors.New("unable to delete vlan")
+	createVLanErrorType   = errors.New("unable to create vlan")
+	createFabricErrorType = errors.New("unable to create fabric")
+	getVlanErrorType      = errors.New("unable to get vlan")
+	getFabricErrorType    = errors.New("unable to get fabric")
+	updateVlanErrorType   = errors.New("unable to update vlan")
+	listVlansErrorType    = errors.New("unable to list vlans")
+	listFabricsErrorType  = errors.New("unable to list fabrics")
+)
+
+func TestDeleteFabric(t *testing.T) {
+	networkClient := MockNetworkClient()
+
+	do := func(ctx context.Context, nc *network.NetworkClient) error {
+		defer testutils.DeactivateClient()
+
+		return nc.Fabrics().Delete(ctx, &network.DeleteFabricInput{
+			FabricVLANID: fakeFabricVlanId,
+			NetworkID:    fakeNetworkId,
+		})
+	}
+
+	t.Run("successful", func(t *testing.T) {
+		testutils.RegisterResponder("DELETE", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks", fakeNetworkId), deleteFabricSuccess)
+
+		err := do(context.Background(), networkClient)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+		testutils.RegisterResponder("DELETE", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId), "networks"), deleteFabricError)
+
+		err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "unable to delete fabric") {
+			t.Errorf("expected error to equal testError: found %v", err)
+		}
+	})
+}
+
+func TestDeleteVLans(t *testing.T) {
+	networkClient := MockNetworkClient()
+
+	do := func(ctx context.Context, nc *network.NetworkClient) error {
+		defer testutils.DeactivateClient()
+
+		return nc.Fabrics().DeleteVLAN(ctx, &network.DeleteVLANInput{
+			ID: fakeFabricVlanId,
+		})
+	}
+
+	t.Run("successful", func(t *testing.T) {
+		testutils.RegisterResponder("DELETE", path.Join("/", accountUrl, "fabrics", "default", "vlans", strconv.Itoa(fakeFabricVlanId)), deleteVLanSuccess)
+
+		err := do(context.Background(), networkClient)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("error", func(t *testing.T) {
+		testutils.RegisterResponder("DELETE", path.Join("/", accountUrl, "fabrics", "default", "vlans"), deleteVLanError)
+
+		err := do(context.Background(), networkClient)
+		if err == nil {
+			t.Fatal(err)
+		}
+
+		if !strings.Contains(err.Error(), "unable to delete vlan") {
```

Comment:
Is this right?  Don't we have a better, more explicit way of testing the error code now that we're in `pkg/errors` ?